hw/ip/otp_ctrl/data/BUILD

```diff
@@ -16,60 +21,310 @@ autogen_hjson_header(
 )
 
 otp_json(
-    name = "otp_ctrl_dev_json",
-    lc_count = 5,
-    lc_state = "DEV",
+    name = "otp_json_base_config",
```

Comment:
Are these translations of the existing hjson config files?  Are we planning on deleting those files at some point in favor of these in-bazel defined configs?

Reply:
@moidx are we ok with moving all these definitions into Bazel rules, or do we want to keep the source of truth in separate Hjson files?

Reply:
+1 to having a single source of truth in the long term so that it is easier to keep test cases in sync across targets (dv, fpga, verilator). 

Moving things to Bazel has the benefit of being able to customize OTP images as part of test target definitions. I believe only ROM E2E tests running on FPGA/verilator are taking advantage of this. I assume that once we start running those tests in DV we will need to add support for this in dvsim.

@msfschaffner can you outline what we would be missing if we move these configs to Bazel? One thing I think it would be good to expose is the seed value as a DV configurable param. Are there any other things we are missing?

Here is some additional context on the DV side:

## Current dv configuration

In the current DV target configuration, OTP images need to be predefined in the dvsim configuration. For example: 

1. `chip_sim_cfg.hjson`: pre-run commands, e.g.

```hjson
{gen_otp_images_cmd} \
              --img-cfg {gen_otp_images_cfg_dir}/otp_ctrl_img_dev.hjson \
              --add-cfg {gen_otp_images_cfg_dir}/otp_ctrl_img_creator_sw_cfg.hjson \
              --add-cfg {gen_otp_images_cfg_dir}/otp_ctrl_img_owner_sw_cfg.hjson \
              --add-cfg {gen_otp_images_cfg_dir}/otp_ctrl_img_hw_cfg.hjson \
              --out {run_dir}/otp_ctrl_img_dev.vmem \
              {gen_otp_images_cmd_opts}
```

4. `hw/top_earlgrey/dv/env/chip_env_cfg.sv`: Mapping to DV UVM params, e.g.

```sv
    otp_images[lc_ctrl_state_pkg::LcStRaw] = "otp_ctrl_img_raw.vmem";
    otp_images[lc_ctrl_state_pkg::LcStDev] = "otp_ctrl_img_dev.vmem";
    otp_images[lc_ctrl_state_pkg::LcStProd] = "otp_ctrl_img_prod.vmem";
    otp_images[lc_ctrl_state_pkg::LcStRma] = "otp_ctrl_img_rma.vmem";
    otp_images[lc_ctrl_state_pkg::LcStTestUnlocked0] = "otp_ctrl_img_test_unlocked0.vmem";
```

5. `chip_sim_cfg.hjson`: Request specific OTP configuration via parameter setting, e.g. `+use_otp_image=LcStTestUnlocked0`

```
    {
      name: chip_sw_csrng_lc_hw_debug_en_test
      uvm_test_seq: chip_sw_csrng_lc_hw_debug_en_vseq
      sw_images: ["//sw/device/tests/sim_dv:csrng_lc_hw_debug_en_test:1"]
      en_run_modes: ["sw_test_mode_test_rom"]
      run_opts: ["+sw_test_timeout_ns=15_000_000", "+rng_srate_value_min=15",
                 "+use_otp_image=LcStTestUnlocked0"]
      run_timeout_mins: 60
    }
```
Note: DV still has the ability to inject OTP values via backdoor, so there is still a chance values will not match across targets. The backdoor is required to optimize simulation time (otherwise it would take took long to run the same tests with the foundry database).

## Moving to Bazel OTP targets

The following could be done to start aligning DV with FPGA build infra. We would have to make changes to DV infra to be able to consume the OTP image target.

1. `chip_sim_cfg.hjson`: Request specific OTP configuration via build target:

```
    {
      name: chip_sw_csrng_lc_hw_debug_en_test
      uvm_test_seq: chip_sw_csrng_lc_hw_debug_en_vseq
      sw_images: ["//sw/device/tests/sim_dv:csrng_lc_hw_debug_en_test:1"]
      otp_images: ["//hw/ip/otp_ctrl/data:otp_test_unlocked0_image"]
      en_run_modes: ["sw_test_mode_test_rom"]
      run_opts: ["+sw_test_timeout_ns=15_000_000", "+rng_srate_value_min=15"]
      run_timeout_mins: 60
    }
```

Note:  Currently, the colon after the sw_image target name (e.g. :1) indicates the image type: 

```sv
  typedef enum {
    SwTypeRom,  // Ibex SW - first stage boot ROM.
    SwTypeTest, // Ibex SW - actual test SW.
    SwTypeOtbn  // Otbn SW.
  } sw_type_e;
```

Adding @sriyerg  since this is getting into DV land. 

If we need to discuss this in more detail, perhaps it can be added to the next top-level tests agenda?

CC @timothytrippel

Reply:
I am not aware of the background for this change. Could you please share the doc? I would expect the way to handle this would be to simply replace the python invocations with bazel invocations, and somehow retain the ability to pass the seed value to the invocation.